portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java

```diff
@@ -48,35 +49,17 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
           return;
       }
 
-      String nickname = getNickname(userService.getCurrentUser().getEmail());
+      String nickname = RetrieveNickname.getNickname(userService);
       if (nickname == null) {
           response.sendRedirect("/nickname.html");
           return;
       } 
 
-      String userEmail = userService.getCurrentUser().getEmail();
       String urlToRedirctToAfterUserLogsOut = "/index.html";
       String logoutUrl = userService.createLogoutURL(urlToRedirctToAfterUserLogsOut);
-      out.println("<p>Hello " + userEmail + "!</p>");
-      out.println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
-      out.println("<p>If you want to change your nickname, click <a href=\"/nickname.html\">here</a>.</p>");
-    }
-
-    public static String getNickname(String email) {
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        Query query = new Query("UserInfo").setFilter(new FilterPredicate("email", FilterOperator.EQUAL, email));
-        PreparedQuery results = datastore.prepare(query);
-        
-        int count = 0;
-        String nickname = null;
-        for (Entity entity: results.asIterable()) {
-            count += 1;
-            if (entity == null) {
-                return null;
-            }
-            nickname = (String) entity.getProperty("nickname");
-        }
-        if (count == 0) return null;
-        return nickname;
+      out.println("<p>Welcome back " + nickname + "!</p>");
```

Comment:
Try setting your nickname to `Leonard<script>alert("hello");</script>` and then load this page :)

We can implement HTML sanitization another time, but for now it's better to not render possible HTML that was provided by the user. So, let's just remove the `nickname` part of this line.